docs/source/frameworks/onnx.rst

```diff
@@ -0,0 +1,329 @@
+
+   .. code-block:: python
+
+      import onnxruntime
+
+      ort_session = onnxruntime.InferenceSession("super_resolution.onnx")
+      # compute ONNX Runtime output prediction
+      ort_inputs = {ort_session.get_inputs()[0].name: to_numpy(x)}
+      # ONNX Runtime will return a list of outputs
+      ort_outs = ort_session.run(None, ort_inputs)
+      print(ort_outs[0])
+
+.. tab:: TensorFlow
+
+   TODO
+
+.. tab:: Scikit Learn
+
+   TODO
+
+
+Saving ONNX model with BentoML
+-----------------------------
+
+To quickly save an ONNX model to BentoML's :ref:`Model
+Store<concepts/model:Managing Models>`, first use ``onnx.load`` to
+load the exported ONNX model back into ``onnx.ModelProto`` object,
+then call BentoML's ``save_model``:
+
+
+.. code-block:: python
+
+   signatures = {
+       "run": {"batchable": True},
+   }
+   bentoml.onnx.save_model("super_resolution", onnx_model, signatures=signatures)
+
+which will result:
+
+.. code-block:: bash
+
+   Model(tag="super_resolution:lwqr7ah5ocv3rea3", path="~/bentoml/models/super_resolution/lwqr7ah5ocv3rea3/")
+
+.. note::
+
+   ``save_model`` will use ``{"run": {"batchable": False}}`` as
+   default signatures if ``signatures`` is not provided. Set
+   ``batchable`` to ``False`` will disable BentoML's
+   :ref:`guides/batching:Adaptive Batching` functionality. That's why
+   we provide our own signatures here. Read more about :ref:`Model
+   Signatures <concepts/model:Model Signatures>` and :ref:`Batch Input
+   <concepts/model:Batching>`
+
+.. seealso::
+
+   ``save_model`` also has some :ref:`general options
+   <concepts/model:Save A Trained Model>` for functionalities like
+   saving metadata and custom objects.
+
+
+Loading an ONNX model with BentoML
+----------------------------------
+
+We can use ``load_model`` to load an ONNX model back to memory:
+
+.. code-block:: python
+
+   ort_session = bentoml.onnx.load_model("onnx_super_resolution")
+
+.. note::
+
+   BentoML will load an ONNX model back as an
+   ``onnxruntime.InferenceSession`` object which is ready to do
+   inference
+
+
+Then we can do the inference:
+
+.. code-block:: python
+
+   test_input = np.random.randn(2, 1, 244, 244) # can accept arbitrary batch size
+   ort_session.run(None, {"input": test_input.astype(np.float32)})
+
+.. note::
+
+   In above codes we need explicitly to convert input ndarray to
+   float32 because ``onnxruntime.InferenceSession`` only expects
+   single floats. In the following section we will see that BentoML
+   runner will automatically cast input data to this type
+
+
+In addition to :ref:`general options <concepts/model:Retrieve a saved
+model>` provided by :code:`load_model`, you can pass ``providers`` to
+set ONNX Runtime's `Execution Providers
+<https://onnxruntime.ai/docs/execution-providers/>`_. You can also
+pass ``session_options`` to set ``onnxruntime.InferenceSession``'s
+`SessionOptions
+<https://onnxruntime.ai/docs/api/python/api_summary.html#sessionoptions>`_.
```

Comment:
I think this can be its own sections about execution providers. And what is the default behaviour for runners with regarding this options.
You can get into what users can expect when a ONNX runner is running on which hardware.